### PR TITLE
Redirect and show error message when diff revision id is not active

### PR DESCRIPTION
**Pull request by praveenkuttappan**
Accessing a URL with diff revision ID throws error if that diff revision is no longer valid. PR has the change to redirect user to review page instead of failing with a server error message.

### CODE REVIEW

**Review by JonathanCrd — APPROVED** [review]
🚀

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml [hunk 1]
```diff
@@ -482,7 +482,12 @@
                     var diffRevisionType = (Model.ReviewContent.APIRevisionsGrouped.ContainsKey("Automatic")) ? "Automatic" : Model.ReviewContent.APIRevisionsGrouped.Keys.First();
                     @if (Model.DiffRevisionId != null)
```
Comment (weshaggard) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml:483]: Should this DiffRevisionId condition be changed like you did below?

Comment (praveenkuttappan) [reply src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml:483]: Yes . I will change that too. Thanks

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml [hunk 2]
```diff
@@ -480,9 +480,14 @@
                 @if (@Model.ReviewContent.APIRevisionsGrouped.SelectMany(kvp => kvp.Value).Count() > 1)
                 {
                     var diffRevisionType = (Model.ReviewContent.APIRevisionsGrouped.ContainsKey("Automatic")) ? "Automatic" : Model.ReviewContent.APIRevisionsGrouped.Keys.First();
-                    @if (Model.DiffRevisionId != null)
+                    @if (Model.ReviewContent.DiffAPIRevision != null)
                     {
-                        diffRevisionType = Model.ReviewContent.DiffAPIRevision.APIRevisionType.ToString();
+                        // Diff revision is null if someone reuses an old URL with a deleted revision as  diff revision
+                        // Users copy and shares diff URL to approvers and sometimes they delete and recreate the revision which makes the diff revision null
+                        if (Model.ReviewContent.DiffAPIRevision?.APIRevisionType != null)
```
Comment (weshaggard) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml:487]:
```suggestion
                        if (Model.ReviewContent.DiffAPIRevision.APIRevisionType != null)
```
Shouldn't need it given you checked for null above.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**